### PR TITLE
Updating extension to accomodate running headless for tests and addin…

### PR DESCRIPTION
**Pull request by LaCuneta**
…g unit tests.

Switched to the extension context for grabbing the reflection info in all primitives so they'll run headless as well as in NetLogo desktop.  Added unit tests for some of the basic use cases for each primitive.

### CODE REVIEW

File: tests.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+reflection-gets-procedures
+  extensions [ reflection ]
+  to-report rev [tacos] report (tacos - 1) end
+  to start let vs 0 show vs end
+  reflection:procedures => [["REV" "REPORTER" "OTPL" ["TACOS"]] ["START" "COMMAND" "OTPL" []]]
+
+reflection-gets-current-procedure
+  extensions [ reflection ]
+  to-report rev [tacos] set tacos (tacos - 1) report reflection:current-procedure end
```
Comment (TheBizzle) [inline tests.txt:45]: [Tacos rule](https://www.youtube.com/watch?v=ljrIFqbmKp8&feature=youtu.be&t=3s)

**Review by TheBizzle — APPROVED** [review]
LGTM

**Review by mrerrormessage — CHANGES_REQUESTED** [review]
Looks good overall. There's a lot of commented-out code that I would like to see removed before merging.

**Comment by LaCuneta** [comment]
@mrerrormessage Thanks for looking it over, I removed the extra comment lines from the extension code.  Let me know if there is anything else!

**Pull request merged**